pkg/apis/pipeline/v1/pipeline_validation_test.go

```diff
@@ -1581,12 +1613,32 @@ func TestValidatePipelineParameterVariables_Failure(t *testing.T) {
 				Params: []Param{{
 					Name: "a-param", Value: ParamValue{Type: ParamTypeArray, ArrayVal: []string{"$(params.foo)"}},
 				}, {
-					Name: "b-param", Value: ParamValue{Type: ParamTypeArray, ArrayVal: []string{"$(params.does-not-exist)"}},
+					Name: "b-param", Value: ParamValue{Type: ParamTypeArray, ArrayVal: []string{"$(params.does-not-exist)"}}}}},
+		}},
+		expectedError: apis.FieldError{
+			Message: `non-existent variable in "$(params.does-not-exist)"`,
+			Paths:   []string{"[0].matrix.params[b-param].value[0]"},
+		},
+	}, {
+		name: "invalid pipeline task with two matrix include parameters and one of them missing from the param declarations",
+		params: []ParamSpec{{
+			Name: "foo", Type: ParamTypeString,
+		}},
+		tasks: []PipelineTask{{
+			Name:    "foo-task",
+			TaskRef: &TaskRef{Name: "foo-task"},
+			Matrix: &Matrix{
+				Include: []MatrixInclude{{
+					Params: []Param{{
+						Name: "a-param", Value: ParamValue{Type: ParamTypeString, StringVal: "$(params.foo)"},
+					}, {
+						Name: "b-param", Value: ParamValue{Type: ParamTypeString, StringVal: "$(params.does-not-exist)"},
+					}},
 				}}},
 		}},
 		expectedError: apis.FieldError{
 			Message: `non-existent variable in "$(params.does-not-exist)"`,
-			Paths:   []string{"[0].matrix[b-param].value[0]"},
+			Paths:   []string{"[0].matrix.include.params[b-param].value[1]"},
```

Comment:
What does the index here refer to? In this example, include.params[1] is named "b-param", and refers to a value that doesn't exist. I think this error message might be hard for users to understand and fix; it might make more sense to return `matrix.include.params[1]` or `matrix.include.params[1].value` instead.

(I think `params[b-param]` is a bit confusing, since params isn't a dict, but probably most people could figure this out, but `value[1]` is very confusing because value is not an array.)